pennylane/pauli/pauli_arithmetic.py

```diff
@@ -231,6 +231,29 @@ def __mul__(self, other):
 
     __rmul__ = __mul__
 
+    def __add__(self, other):
+        """Add PauliWords/Sentences"""
+        # Note that the case of PauliWord + PauliSentence is covered in PauliSentence
+        if isinstance(other, PauliWord):
+            if other == self:
+                return PauliSentence({self: 2.0})
+            return PauliSentence({self: 1.0, other: 1.0})
+
+        if isinstance(other, TensorLike):
+            # Scalars are interepreted as scalar * Identity
+            IdWord = PauliWord({})
+            if IdWord == self:
+                return PauliSentence({self: 1.0 + other})
+            return PauliSentence({self: 1.0, IdWord: other})
+
+        return NotImplemented
+
+    __radd__ = __add__
+
+    def __iadd__(self, other):
+        """Inplace addition of PauliWords"""
+        return self + other
```

Comment:
I'm nor sure `__iadd__` makes sense becaue pauli words are immutable and can't be in place modified.

Reply:
I was also wondering if that is a problem conceptually because the returned object is not even a PauliWord anymore. The motivation to have it anyway comes from building operators iteratively like so
```
H = PauliWord({0:"X"})
H += PauliWord({1:"X"})
H += PauliWord({0:"X"}) @ PauliWord({1:"X"})
```

Reply:
Let's delete `__iadd__`, as we can't "modify" an immutable object.  

We could do the syntax you requested simply by starting with a `PauliSentence`:
```
H = 1 * PauliWord({0:"X"})
H += PauliWord({1:"X"}) # PauliSentence.__iadd__
H += PauliWord({0:"X"}) @ PauliWord({1:"X"}) # PauliSentence.__iadd__
```

Reply:
Ignore that.  I just looked up the Python docs: https://docs.python.org/3/library/operator.html#in-place-operators

```
For immutable targets such as strings, numbers, and tuples, the updated value is computed, but not assigned back to the input variable:
```

Didn't know that. We're good here.